_posts/2022-12-01-trino-summit-2022-sk-telecom-recap.md

```diff
@@ -0,0 +1,98 @@
+observable federated data platform on open source software at scale. 
+
+SK Telecom manages on-premise Hadoop clusters to store their data. SK Telecom
+started using Presto in 2016, and shifted to Trino in 2021. To run batch queries
+on their warehouse, Trino workers are deployed on HDFS data nodes. There is also
+an adhoc Trino cluster deployed to manage federated queries over multiple data
+silos from an array of disparate data sources. Previously, they used tools like 
+[distcp](https://hadoop.apache.org/docs/stable/hadoop-distcp/DistCp.html) to
+make data available in one center. This was one of the slow and brittle process
+that Trino replaced. They chose Trino because it simplified querying novel big
+data systems and combining that data more commonplace systems for their users.
+
+As Trino adoption grew within the company to the tune of 300 requests per
+minute, they eventually faced challenges scaling. Not only were the number of
+requests growing, but users increased the range of data they wanted to query.
+The total data size they were evaluating was petabytes in size, involving
+terabyte size query input. This was all bing processed over hundreds of nodes.
+Many users were queued in a blocked state while waiting for resources to become
+available. At this point, Trino query performance also began to suffer. It was
+at this point the data engineering team began investigating how they could both
+scale and improve individual query performance.
+
+To find the root cause, they had to expand beyond the web UI. They began
+collecting all result json files, coordinator and worker jmx stats, system
+metrics, and Trino logs to build out their own metrics dashboard. The two main
+causes was that input data was too large, and they saw spikes in the number of
+`BlockedSplit` operations that lead to queries waiting for other tasks to
+complete and in a blocked state. They initially aimed to address this by
+updating some settings to increase thread counts and tuning queries, but this
+still didn't achieve the desired results. The ultimate bottleneck was the Hive
+metastore and the expensive list operations that caused many of the blocked
+operations to finish slowly.
+
+At this point, the team reevaluated their needs to consider alternative solution
+spaces. They needed a better indexing strategy on the data Hive data with a
+flexible partitioning strategy. They also needed to remove the bottleneck on the
+metadata for this data while still maintaining compatibility across multiple
+query engines as Hive did for the most part.
+
+The team looked at the existing set of novel data lake connectors available in
+Trino version 356, which at the time only included Iceberg. SK Telecom was 
+immediately impressed by the metadata indexing in the Iceberg project. They 
+particularly liked the snapshot isolation as new data was created or existing 
+data is modified. They were able to speed up queries using data file pruning
```

Comment:
"new data was created or existing data is modified" -> "data is created or modified"